### PR TITLE
New  minor bugs fixed and background color changed.

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
Description:
Now if you divide by zero you get an error message.
Now if you press the "=" key, without having any value, it returns zero (it no longer crashes).
Now if you press the "=" key, having only a value, it returns that number (it no longer crashes).
Now if you press an operator and then the = key, you get an error message (it no longer crashes).
Now if the screen is rotated, the values and operations ​​are no longer lost.
Now the background color of the calculator has been changed to match the palette used.

image

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,13 +9,13 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.MyCalculator">
-        <activity android:name=".MainActivity">
+        <activity
+            android:name=".MainActivity"
+            android:configChanges="orientation|screenSize">
```
Comment (SartiMau) [inline app/src/main/AndroidManifest.xml:14]: why do you add this line?

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 2]
```diff
@@ -28,52 +29,69 @@ public void saveNumber(String number) {
...
     @Override
     public String getPartialResult() {
-
         return (firstOperand + operator + secondOperand);
     }
 
+    private boolean isValidOperation(){
+        if (operator.isEmpty()) {
+            if (firstOperand.isEmpty()){
+                result = "0";
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: use a constant for this pls

File: app/src/main/java/com/example/mycalculator/utils/StringUtils.java [hunk 3]
```diff
@@ -7,5 +7,7 @@
     public static final String OPERATOR_MINUS = "-";
     public static final String OPERATOR_DIVIDE = "/";
     public static final String OPERATOR_MULTIPLY = "X";
+    public static final String ERROR_MESSAGE_DIVISION = "Error, can not divide by zero";
+    public static final String ERROR_MESSAGE_INVALID_FORMAT = "Error: Invalid format";
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/utils/StringUtils.java]: do you use these strings to show them on the UI? if so, you need to add it on the String res, not to a constant. Check the error handling

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 4]
```diff
@@ -28,52 +29,69 @@ public void saveNumber(String number) {
...
-                break;
+                case OPERATOR_MINUS:
+                    result = String.valueOf(Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand));
+                    break;
 
-            case OPERATOR_DIVIDE:
-                result = String.valueOf(Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand));
-                break;
+                case OPERATOR_DIVIDE:
+                    if ((Double.parseDouble(secondOperand)) == Double.parseDouble("0")) {
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: you have a harcode here

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Matias-Villalobo) [reply app/src/main/AndroidManifest.xml:14]: I used that property since it allowed me that when rotating the phone, the values ​​of the operation would not be lost.

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 2 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done.

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 4 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -9,13 +9,13 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.MyCalculator">
-        <activity android:name=".MainActivity">
+        <activity
+            android:name=".MainActivity">
+            //android:configChanges="orientation|screenSize">
```
Comment (mauroLonghi) [inline app/src/main/AndroidManifest.xml]: remove this comment line

Comment (Matias-Villalobo) [reply app/src/main/AndroidManifest.xml]: Done

**Comment by Matias-Villalobo** [comment]
Here you can see how the values ​​are maintained even if the screen is rotated


https://user-images.githubusercontent.com/73836200/103543174-d4eac900-4e7c-11eb-951a-b959773cc19f.mp4

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (mauroLonghi) [reply app/src/main/AndroidManifest.xml:14]: yes its a valid use of the changes of the screen https://developer.android.com/guide/topics/resources/runtime-changes#:~:text=Possible%20values%20are%20listed%20in,when%20the%20screen%20orientation%20changes.

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 6]
```diff
@@ -28,52 +32,69 @@ public void saveNumber(String number) {
...
-        operator = EMPTY_STRING;
-        return EMPTY_STRING;
-    }
+        @Override
+        public String eraseResult () {
+            result = EMPTY_STRING;
+            firstOperand = EMPTY_STRING;
+            secondOperand = EMPTY_STRING;
+            operator = EMPTY_STRING;
+            return EMPTY_STRING;
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: why in this method you return a String? is really necessary?

File: app/src/main/java/com/example/mycalculator/utils/StringUtils.java (hunk 3 above)
Comment (LuchoElissondoGlobant) [reply app/src/main/java/com/example/mycalculator/utils/StringUtils.java]: matias, you could use an enum class with error codes, and then map the errors..

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 6 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: i used a string since all values ​​are of that type, and then parsed to double. What I am looking for is that when pressing the delete key, all the values ​​return to the initial state.

**Comment by Matias-Villalobo** [comment]
In this video you can see how the calculator now performs combined operations as well as negative values ​​can also be used.
(Those were the cases that were not shown in the previous demo/gif)

https://user-images.githubusercontent.com/73836200/103892150-d52bd400-50c9-11eb-89aa-78955c6f8b5f.mp4

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 7]
```diff
@@ -2,62 +2,142 @@
...
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand = EMPTY_STRING;
-    private String secondOperand = EMPTY_STRING;
+    private Operand firstOperand = new Operand();
+    private Operand secondOperand = new Operand();
     private String operator = EMPTY_STRING;
     private String result = EMPTY_STRING;
 
+    private class Operand {
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: this Class must move in other file, please change it. Also replace all the hardcode please

Comment (lucianoeli) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: You shouldnt create an inner class, you could create this one in a separete file

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 8]
```diff
@@ -2,62 +2,142 @@
...
+        private String sign = "";
```
Comment (lucianoeli) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: please check the harcodes of this file

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 7 above)
Comment (lucianoeli) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: you should create it in a separete file and inject them in the constructor, if not you would not be able to test

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 9]
```diff
@@ -2,62 +2,142 @@
...
+                    break;
+
+                case OPERATOR_MULTIPLY:
+                    result = String.valueOf(firstOperand.getValue() * secondOperand.getValue());
+                    break;
+
+                default:
+                    result = ERROR_MESSAGE;
+            }
+            if (result.substring(0).equals("-")) {
```
Comment (lucianoeli) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: hardcodes here too.

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 10]
```diff
@@ -66,14 +146,20 @@ public String getFullResult() {
...
+        secondOperand.eraseOperands();
         operator = EMPTY_STRING;
         return EMPTY_STRING;
     }
 
     @Override
     public void saveOperationSymbol(String operatorSymbol) {
-        operator = operatorSymbol;
+        if (firstOperand.isEmpty()) {
+            firstOperand.setSign("-");
```
Comment (lucianoeli) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: check this hardcodes

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 7 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done, another class created.

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 8 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 9 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 10 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+    public String sign = EMPTY_STRING;
+    public String value = EMPTY_STRING;
+
+    private String getSign() {
+        return sign;
+    }
+
+    public void eraseOperands() {
+        sign = "";
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/Operand.java]: still harcodessss

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        value = "";
+    }
+
+    public Double getValue() {
+        if (value == EMPTY_STRING) {
+            return 0.0;
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/Operand.java]: ssame here

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
+package com.example.mycalculator.mvp;
+
+import static com.example.mycalculator.utils.StringUtils.EMPTY_STRING;
+
+public class Operand {
+
+    public String sign = EMPTY_STRING;
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/Operand.java]: you have already defined a EMPTY_STRING in StringUtils... please use that one

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 14]
```diff
@@ -129,11 +85,11 @@ public String getFullResult() {
                 default:
                     result = ERROR_MESSAGE;
             }
-            if (result.substring(0).equals("-")) {
-                firstOperand.sign = "-";
+            if (result.substring(0).equals(OPERATOR_MINUS)) {
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: still hardcodes here

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 12 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 11 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 13 above)
Comment (lucianoeli) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java:7]: correct. my bad.

Comment (LuchoElissondoGlobant) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: my bad. this is correct

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 12 above)
Comment (LuchoElissondoGlobant) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: did you? i still see the 0.0

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 11 above)
Comment (LuchoElissondoGlobant) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: did you commit and push?

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 12 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: That will be fixed in the next commit

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 11 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: Not yet

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 14 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    public Boolean isEmpty() {
+        return value.isEmpty();
+    }
+
+    public String getFullResult() {
+        return null;
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/mycalculator/mvp/Operand.java]: this method it used? and why you return a null?

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 16]
```diff
@@ -11,53 +16,79 @@
...
+                case OPERATOR_SUM:
+                    result = String.valueOf(firstOperand.getValue() + secondOperand.getValue());
+                    break;
+
+                case OPERATOR_MINUS:
+                    result = String.valueOf(firstOperand.getValue() - secondOperand.getValue());
+                    break;
+
+                case OPERATOR_DIVIDE:
+                    if (firstOperand.getValue() == secondOperand.getValue()) {
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: this condition are right? because if you try to do 7 / 7 = 1

File: app/src/main/java/com/example/mycalculator/utils/NumbersUtils.java [hunk 17]
```diff
@@ -0,0 +1,7 @@
+package com.example.mycalculator.utils;
+
+public class NumbersUtils {
+    public static final double DOUBLE_ZERO_NUMBER = 0d;
+    public static final int INT_ZERO_NUMBER = 0;
+    public static final int INT_ONE_NUMBER = 1;
```
Comment (nicolbidone) [inline app/src/main/java/com/example/mycalculator/utils/NumbersUtils.java]: please give constant names a meaning

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 18]
```diff
@@ -11,53 +16,79 @@
...
+            firstOperand.addNumber(number);
         } else {
-            secondOperand += number;
+            secondOperand.addNumber(number);
         }
     }
 
     @Override
     public String getPartialResult() {
+        return String.valueOf(firstOperand.getValue()) + operator + String.valueOf(secondOperand.getValue());
```
Comment (nicolbidone) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java:35]: you can create a function from operand that already returns String value

**Review by nicolbidone — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 15 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: You were right, I was not using those methods, so they were deleted

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 16 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: You were right, now its fixed

File: app/src/main/java/com/example/mycalculator/utils/NumbersUtils.java (hunk 17 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/utils/NumbersUtils.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 19]
```diff
@@ -1,79 +1,142 @@
...
+            error = ErrorUtils.NONE;
+            secondOperand.eraseOperands();
         }
+    }
 
+    public String getResult() {
         return result;
     }
 
+    private void checkNumberSign() {
```
Comment (nicolbidone) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: the name don't describe the function, thisshould be updateFirstOperand

File: app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java [hunk 20]
```diff
@@ -26,7 +26,26 @@ public void operationSymbolPressed(String operatorUsed) {
...
     @Override
     public void operatorResultPressed() {
-        view.drawNumber(model.getFullResult());
+        model.doOperations();
+        switch (model.getError()) {
+            case NONE: {
+                view.drawNumber(model.getResult());
+                break;
+            }
+
```
Comment (nicolbidone) [inline app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java]: please remove this empty line

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 19 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java (hunk 20 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java [hunk 21]
```diff
@@ -0,0 +1,36 @@
+package com.example.mycalculator.mvp;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/Operand.java]: this class should not be here in this package

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java [hunk 22]
```diff
@@ -1,79 +1,142 @@
...
+            secondOperand.addNumber(number);
         }
     }
 
-    @Override
-    public String getPartialResult() {
+    public String getValue() {
+        String firstValue = String.valueOf(firstOperand.getValue());
+        String secondValue = String.valueOf(secondOperand.getValue());
+        String operatorUsed = operator;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: you can avoid the use of this variable and use operator

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java [hunk 23]
```diff
@@ -17,5 +19,19 @@ public CalculatorView(MainActivity mainActivity, ActivityMainBinding binding) {
     public void drawNumber(String number) {
         binding.screenTextExample.setText(number);
     }
+    @Override
+    public void showErrorDivision(String errorMessage) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: you are not using the parameter variable, remove it pls

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java [hunk 24]
```diff
@@ -17,5 +19,19 @@ public CalculatorView(MainActivity mainActivity, ActivityMainBinding binding) {
...
+        binding.screenTextExample.setText(getContext().getString(R.string.error_message_in_division));
+    }
+    @Override
+    public void showErrorInvalidOperation(String errorMessage) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: you are not using the parameter variable, remove it pls

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java [hunk 25]
```diff
@@ -17,5 +19,19 @@ public CalculatorView(MainActivity mainActivity, ActivityMainBinding binding) {
...
+        binding.screenTextExample.setText(getContext().getString(R.string.error_message_when_invalid_format));
+    }
+    @Override
+    public void showErrorMessage(String errorMessage) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: you are not using the parameter variable, remove it pls

File: app/src/main/res/values/strings.xml [hunk 26]
```diff
@@ -16,4 +16,9 @@
     <string name="main_screen_btn_text_split">/</string>
     <string name="main_screen_btn_text_multiplication">X</string>
     <string name="main_screen_btn_text_result">=</string>
+    <string name="error_message_in_division">Error, can not divide by zero</string>
+    <string name="error_message_when_invalid_format">Error: Invalid format</string>
+    <string name="error_generic_message">Error, unknown Symbol used</string>
+
+
```
Comment (SartiMau) [inline app/src/main/res/values/strings.xml]: remove this extra line pls

File: app/src/main/java/com/example/mycalculator/utils/ErrorUtils.java [hunk 27]
```diff
@@ -0,0 +1,10 @@
+package com.example.mycalculator.utils;
+
+    public enum ErrorUtils {
+        NONE,
+        ERROR_MESSAGE_DIVISION,
+        ERROR_MESSAGE_INVALID_FORMAT,
+        ERROR_MESSAGE
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/utils/ErrorUtils.java]: remove this extra line pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/utils/ErrorUtils.java (hunk 27 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/utils/ErrorUtils.java]: Done

File: app/src/main/res/values/strings.xml (hunk 26 above)
Comment (Matias-Villalobo) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java [hunk 28]
```diff
@@ -17,5 +19,19 @@ public CalculatorView(MainActivity mainActivity, ActivityMainBinding binding) {
...
+        binding.screenTextExample.setText(getContext().getString((R.string.error_generic_message)));
+    }
+
```
Comment (lucianoeli) [inline app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: remove extra lines here

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 25 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 24 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 23 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java (hunk 22 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/Operand.java (hunk 21 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: Where do you recommend i should put it?

Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/Operand.java]: in a utils package or in an entity package outside the mvp

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 25 above)
Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: this still here

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 24 above)
Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: this still here

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 23 above)
Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: this still here

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Comment by Matias-Villalobo** [comment]
All previous comments have been resolved. As can be seen in the video, the UI has also been modified, so that only the number entered is displayed. Also in the video you can see operations combined with different signs.


https://user-images.githubusercontent.com/73836200/104218107-08989680-541b-11eb-9eec-4b2b066de262.mp4

File: app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java [hunk 29]
```diff
@@ -26,12 +26,35 @@ public void operationSymbolPressed(String operatorUsed) {
...
+        }
     }
 
     @Override
     public void erase() {
         view.drawNumber(model.eraseResult());
     }
 
+    @Override
+    public void showErrorMessage() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java]: why is this here??

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java [hunk 30]
```diff
@@ -18,4 +20,18 @@ public void drawNumber(String number) {
         binding.screenTextExample.setText(number);
     }
 
+    @Override
+    public void showErrorDivision() {
+        binding.screenTextExample.setText(getContext().getString(R.string.error_message_in_division));
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: remember to add a null check for the getContext()

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java (hunk 30 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/view/CalculatorView.java]: Done

File: app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java (hunk 29 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/presenter/CalculatorPresenter.java]: It was an error, now its gone

**Review by mauroLonghi — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]
